3-reveal/migrations/6-MDA/Zambia-2020/deploy/mda_dispense_tasks.psql

```diff
@@ -0,0 +1,221 @@
+-- Deploy reveal_mda_zambia_2020:mda_dispense_tasks to pg
+
+BEGIN;
+
+--SET search_path TO :"schema",public;
+
+-- Compute single-jurisdiction metrics
+CREATE MATERIALIZED VIEW ntd_dispense_task_metrics AS
+WITH metrics AS
+(
+    SELECT
+        tasks.plan_identifier AS plan_id,
+        tasks.group_identifier AS jurisdiction_id,
+        COALESCE(COUNT(latest_events.id) FILTER (WHERE latest_events.form_data->>'nPzqDistribute' IN ('Yes')), 0) AS nPzqDistribute,
+        COALESCE(COUNT(latest_events.id) FILTER (WHERE latest_events.form_data->>'availableForTreatment' IN ('Yes')), 0) AS availableForTreatment,
+        COALESCE(SUM(COALESCE(latest_events.form_data->>'nPzqDistributedQuantity', '0')::DECIMAL), 0) AS nPzqDistributedQuantity
+    FROM tasks
+    LEFT JOIN LATERAL
+    	(
+    		SELECT
+    			events.*
+    		FROM events
+    		WHERE
+    			events.task_id = tasks.identifier
+    			AND events.event_type = 'mda_dispense'
+    		ORDER BY events.server_version DESC
+    		LIMIT 1
+    	) AS latest_events
+    	ON TRUE
+    WHERE
```

Comment:
Can we also restrict to only active plans here?

When a plan is marked in active, are all of the tasks w/that plan id marked as inactive? or if I wanted to restrict tasks to only those in active plans would I have to restrict to only active plan IDs?

Reply:
you'd have to join with plans - which is probably an ok way to start the chain here too.

Reply:
or we could have a subquery that get's all the active plan IDs and uses `IN`, which I bet would be slower, but idk I'm also not a pg expert at all